src/passes/J2CLItableMerging.cpp

```diff
@@ -0,0 +1,365 @@
+/*
+ * Copyright 2024 WebAssembly Community Group participants
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// This is a J2CL specific pass that merges itables into vtables. It is meant
+// to be run at the beginning before structs corresponding to Java classes are
+// optimized.
+//
+// Embeds itables into vtables to reduce memory usage.
+
```

Comment:
Adding an example of what the pass does would be good here, in some sort of pseudocode, just to give the general idea of what fields is moved where.

Reply:
Added a more detailed explanation. I think this pass could be generalized, moving (deep) immutable fields in a struct to a struct that is on an immutable field as long as you can prove that there is a injective relation in the initialization of such fields; but I don't think there will be many (if any) other opportunities than what the pass is currently doing.

Reply:
@kripken 
(Fixing the tagging)

Reply:
@gkdn That high-level description makes sense to me too. I don't feel strongly between them.